docs/design/features/lsra-detail.md

```diff
@@ -907,19 +924,162 @@ Future Extensions and Enhancements
+
+  - Set aside a subset of the registers for pre-allocation. This might include
+    all of the callee-save registers, and a small number of caller-save registers.
+    Some tuning would likely be required to determine exactly what and how many to reserve.
+
+  - After `buildIntervals()` has completed,for each register candidate local `Interval`,
+    in sorted order, assign a register that corresponds to its `registerPreferences`. If such
+    a register is not available, stop pre-allocating. This becomes its `assignedInterval`
+    (which will be inactive at the start of alloaction).
+
+  - During `allocateRegisters()` as each new block is encountered, instead of simply copying
+    the predecessor `varToRegMap`, iterate over the most frequently lclVars in the union of the
+    live-in, uses and defs, and displace any `Intervals` that are occupying registers that
+    would be more profitably used by the high-frequencly lclVars, weighing spill costs.
+ 
+### <a name="avoid-split"/>Avoid Splitting Loop Backedges
+
+This is captured as Issue [\#16857](https://github.com/dotnet/coreclr/issues/16857).
+
+When the register allocator performs resolution across block boundaries, it may split critical
+edges (edges from a block with multiple successors to a block with multiple predecessors).
+This can seriously impact performance, and is especially bad when the edge is a loop backedge.
+
+One option would be go avoid edge splitting altogether. Instead, we would construct `CriticalEdgeSet`s that
+would include the set of critical edges that have any edges in common.
+For these, once we have allocated the first block that contains either an in-edge or an out-edge in such
+a set, the variable locations on that edge would be fixed, and future blocks that have edges in that
+set would ensure that the variable locations match. This would eliminate not just backedge splitting,
+but all the extra branches currently inserted for resolution. It remains to be seen whether this would
+outweigh the impact of cases where more resolution moves would be required.
+
+### Avoid Spill When Stack Copy is Valid
+
+The idea here is to avoid spilling at a use if the value on the stack is already the correct value.
+
+Issues that this might address include [\#11344] Spill single-def vars at def, [\#7665] Improve spill placement,
+and [\#7465] Avoiding reg spill to memory when reg-value is consistent with memory.
+
+Currently the register allocator doesn't track whether a local variable has the same value on the stack
+as in a register. The work-in-progress to support "write-thru" EH variables (variables live across exception
+boundaries) adds capability to liveness analysis and code generation (in addition to the register allocator)
+to handle variables that are live in both registers and on the stack. This support could be further leveraged
+to avoid re-storing single-def variables if they are spilled at their definition.
```

Comment:
Does `re-storing` in this context mean loading from memory or storing to memory?

Reply:
It means storing them to memory - I should have said "at a use". I'll try to clarify.